frontend/src/components/SingleFileSelector.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div
+    class="cursor-pointer transition-all hover:bg-gray-100"
+    :class="[state.dropAreaActive ? 'bg-gray-300 opacity-100' : '']"
+    @click="onUploaderClick"
+    @drop="onFileDrop"
+    @dragover.prevent
+    @dragenter="state.dropAreaActive = true"
+    @dragleave="state.dropAreaActive = false"
+  >
+    <input
+      ref="uploader"
+      name="uploader"
+      type="file"
+      :accept="supportFileExtensions.join(',')"
+      class="sr-only hidden"
+      @input="onFileChange"
+    />
+    <slot></slot>
+  </div>
+</template>
+
+
+<script lang="ts" setup>
+import { ref, reactive, PropType } from "vue";
+import { useStore } from "vuex";
+import { useI18n } from "vue-i18n";
+
+interface LocalState {
+  dropAreaActive: boolean;
+}
+
+const props = defineProps({
+    maxFileSizeInMiB: {
+      required: true,
+      type: Number,
+    },
+    supportFileExtensions: {
+      required: true,
+      type: Object as PropType<string[]>,
+    },
+});
+
+const emit = defineEmits(["on-select"]);
+
+const store = useStore();
+const { t } = useI18n();
+
+const state = reactive<LocalState>({
+  dropAreaActive: false,
+});
+
+const uploader = ref<HTMLInputElement | null>(null);
+
+const onUploaderClick = () => {
+  uploader.value?.click();
+}
+
+const onFileChange = () => {
+  const files: File[] = (uploader.value as any).files;
+  if (files && files.length > 0) {
+    const file = files[0];
+    if (!validFile(file)) {
+      return;
+    }
+
+    emit("on-select", file);
+  }
+};
+
+const onFileDrop = (e: any) => {
+  e.preventDefault();
+
+  state.dropAreaActive = false;
+  const files: File[] = e.dataTransfer.files;
+  if (!files.length) {
+    return;
+  }
+
+  const file = files[0];
+  if (!validFile(file)) {
```

Comment:
Is that possible (and necessary) to check `file.type` (MIME type)?

Reply:
As we already check the file extension, do we still need to check the MIME type?